algebra/group/coset.lean

```diff
@@ -0,0 +1,174 @@
+import algebra.group.subgroup
+
+open set
+
+universe u
+variable {G : Type u}
+
+def lcoset [has_mul G] (a : G) (S : set G) : set G := image (λ x, a * x) S
+def rcoset [has_mul G] (S : set G) (a : G) : set G := image (λ x, x * a) S
+
+namespace coset_notation
+
+infix * := lcoset
+infix * := rcoset
+
+end coset_notation
+
+section coset_mul
+open coset_notation
+variable [has_mul G]
+
+lemma mem_lcoset {S : set G} {x : G} (a : G) (hxS : x ∈ S) : a * x ∈ a * S :=
+mem_image_of_mem (λ b : G, a * b) hxS
+
+lemma mem_rcoset {S : set G} {x : G} (a : G) (hxS : x ∈ S) : x * a ∈ S * a :=
+mem_image_of_mem (λ b : G, b * a) hxS
+
+def lcoset_equiv (S : set G) (a b : G) := a * S = b * S
+
+lemma lcoset_equiv_rel (S : set G) : equivalence (lcoset_equiv S) := 
+    mk_equivalence (lcoset_equiv S) (λ a, rfl) (λ a b, eq.symm) (λ a b c, eq.trans)
```

Comment:
style: 0 indent

Reply:
Fixed in https://github.com/leanprover/mathlib/pull/94/commits/91693ec5a4d0762766254c8a4d3153659349e094.